ipam/allocator.go

```diff
@@ -264,6 +244,27 @@ func (alloc *Allocator) free(ident string) error {
 	return <-errChan
 }
 
+// Free (Sync) - release single IP address for container
+func (alloc *Allocator) Free(ident string, addrToFree address.Address) error {
+	errChan := make(chan error)
+	alloc.actionChan <- func() {
+		addrs := alloc.owned[ident]
+		for i, ownedAddr := range addrs {
+			if ownedAddr == addrToFree {
+				alloc.debugln("Freed", addrToFree, "for", ident)
+				addrs = append(addrs[:i], addrs[i+1:]...)
+				alloc.owned[ident] = addrs
```

Comment:
Should this delete the entry when `len(addrs) == 0`? (or `addr == nil`; I can never remember whether these two are equivalent in go).